velox/expression/VectorWriters.h

```diff
@@ -511,7 +511,7 @@ struct VectorWriter<Generic<T, comparable, orderable>>
   struct isRowWriter<writer_ptr_t<Row<F...>>> : public std::true_type {};
 
   void finalizeNull() override {
-    if (castType_) {
+    if (writer_.casted_) {
```

Comment:
Using the custom flags is to fix the bugs where `commit()` and `finish()` is called even if the GenericWriter is not casted.https://github.com/facebookincubator/velox/pull/6116

The reason behind the bugs is that `ensureWriter` is called not only during cast, but also sometimes in `init`. As a result, `castType_` will be set even if no any cast is done. 

It will also be a problem here (and also in https://github.com/facebookincubator/velox/pull/6116) since I am directly write to the base vector without casting to any types. So `commit` and `finish` should not be called. Using custom flags solve the bugs